src/lib.rs

```diff
@@ -102,25 +99,27 @@ impl Readings for Nau7802{
 
 impl SensorT<f64> for Nau7802 {
     fn get_readings(&self) -> Result<TypedReadingsResult<f64>, SensorError> {
-    let reading = self.read_adc();
-    let mut x = HashMap:new();
-    x.insert("raw_data".to_string(), reading as f64); // does this even work, do I have to convert?
-    log::debug!("getting raw data from nau7802 succeeded!")
-    Ok(x)
+        let reading = self.read_adc();
+        let mut x = HashMap::new();
+        x.insert("raw_data".to_string(), reading as f64); // does this even work, do I have to convert?
+        //log::debug!("getting raw data from nau7802 succeeded!");
+        Ok(x);
     }
 }
 
-pub fn read_adc(&mut self) -> Result<i32, E> { // i32 -> f64 needed
-    let mut data = [u8; 3];
-    self.i2c_handle.write_read(NAU7802_I2C_ADDRESS, &[NAU7802_REG_ADC], &mut data)?;
-    let raw_value = ((data[0] as i32) << 16) | ((data[1] as i32) << 8) | (data[2] as i32);
+impl Nau7802{
+    pub fn read_adc(self) -> Result<i32, AnalogError> { // i32 -> f64 needed
```

Comment:
I don't think we want an Analog error here, we're just reading from an i2c bus and returning floats, is sensor error a better choice?

Reply:
sure! I wasn't 100% sure which error to pick